subsys/shell/shell_utils.c

```diff
@@ -498,15 +498,16 @@ void z_shell_cmd_trim(const struct shell *sh)
 
 static const struct device *shell_device_internal(size_t idx,
 						  const char *prefix,
-						  shell_device_filter_t filter)
+						  shell_device_filter_t filter,
+						  bool ready)
```

Comment:
```suggestion
						  bool show_all)
```
If you flip the logically intention here, I think the conditional later on simplifies.
Alternatively keep the current logic, but then I suggest renaming to something like `only_ready` to show the intended use a bit more from the get-go.

Reply:
I first agreed on this, and then I remembered that I wanted a way to auto-complete on non-ready devices... which means I miss a function to do so like shell_device_lookup_non_ready() of some sort.

So looks like I forgot that feature at some point

Reply:
Actually what I did is pretty stupid...